tests/test_environment_paths.py

```diff
@@ -0,0 +1,197 @@
+        # repo.
+        self.setup_fixtures()
+
+        # Update the mocked hierarchy to contain the user folder on Linux.
+        if sys.platform == "linux2":
+            full_path = os.path.expanduser("~")
+            current_folder = self._linux_mock_hierarchy
+            for t in self._recursive_split(full_path):
+                current_folder[t] = {}
+                current_folder = current_folder[t]
+            current_folder.update(self._linux_mock_hierarchy["usr"]["local"])
+
+    def test_nuke_path_append(self):
+        """
+        Tests the generation of NUKE_PATH environment variable , to make
+        sure that the toolkit startup paths for Nuke are correctly appended to any pre-existing paths.
+        Checks against pre-existing and non-existent scenarios.
+        :return: None
+        """
+        nuke_path_1 = os.path.join(tempfile.gettempdir(), "gizmo_1")
+        nuke_path_2 = os.path.join(tempfile.gettempdir(), "gizmo_1")
+        plugin_path = os.path.join(repo_root, "plugins", "basic")
+
+        try:
+            # test override
+            os.environ["NUKE_PATH"] = os.pathsep.join([nuke_path_1, nuke_path_2])
+
+            # create launcher
+            nuke_launcher = sgtk.platform.create_engine_launcher(
+                self.tk,
+                sgtk.context.create_empty(self.tk),
+                "tk-nuke",
+                ["10.0v5"]
+            )
+
+            # generate launch env
+            launch_info = nuke_launcher.prepare_launch("/path/to/nuke", ["arg1", "arg2"], None)
+
+            # ensure that the nuke path was preserved and placed first in the path
+            self.assertEqual(
+                launch_info.environment["NUKE_PATH"],
+                os.pathsep.join([nuke_path_1, nuke_path_2, plugin_path])
+            )
+
+        finally:
+            # reset our NUKE_PATH
+            del os.environ["NUKE_PATH"]
+
+        # now that NUKE_PATH is clear, test without stuff in the nuke path
+
+        # generate launch env
+        launch_info = nuke_launcher.prepare_launch("/path/to/nuke", ["arg1", "arg2"], None)
+
+        # ensure that the nuke path was preserved and placed first in the path
+        self.assertEqual(
+            launch_info.environment["NUKE_PATH"],
+            plugin_path
+        )
+
+    def test_hiero_path_append(self):
+        """
+        Tests the generation of HIERO_PLUGIN_PATH environment variable , to make
+        sure that the toolkit startup paths for Hiero are correctly appended to any pre-existing paths.
+        Checks against pre-existing and non-existent scenarios.
+        :return: None
+        """
+        nuke_path_1 = os.path.join(tempfile.gettempdir(), "gizmo_1")
+        nuke_path_2 = os.path.join(tempfile.gettempdir(), "gizmo_1")
+        plugin_path = os.path.join(repo_root, "plugins", "basic")
+
+        try:
+            # test override
+            os.environ["HIERO_PLUGIN_PATH"] = os.pathsep.join([nuke_path_1, nuke_path_2])
```

Comment:
Using [tank_test.temp_env_var](https://github.com/shotgunsoftware/tk-core/blob/master/tests/python/tank_test/tank_test_base.py#L103) instead to set and reset env var.